### PR TITLE
Add isSelected property

### LINKED ISSUE #905: ActionButton should support is-selected
# 🙋 Feature Request

According to https://opensource.adobe.com/spectrum-css/actionbutton.html there is an "is-selected" property which is not available through SpectrumActionButtonProps.

## 🤔 Expected Behavior

SpectrumActionButtonProps has a property "isSelected"

## 😯 Current Behavior

No property is available. Current work around is to use `UNSAFE_className="is-selected"` but that feels wrong

## 💁 Possible Solution

Add the property

## 🔦 Context

I'd like to have a toolbar-button who has a "toggle"-button behavior

### PR DESCRIPTION
**Pull request by tomsontom**
Closes #905

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-types/button/src/index.d.ts [hunk 1]
```diff
@@ -62,6 +62,7 @@ export interface SpectrumButtonProps extends AriaBaseButtonProps, ButtonProps, L
 export interface SpectrumActionButtonProps extends AriaBaseButtonProps, ButtonProps, StyleProps {
   /** Whether the ActionButton should be displayed with a [quiet style](https://spectrum.adobe.com/page/action-button/#Quiet). */
   isQuiet?: boolean
```
Comment (mischnic) [inline packages/@react-types/button/src/index.d.ts:64]:
```suggestion
  isQuiet?: boolean,
```

**Comment by devongovett** [comment]
Hey @tomsontom, thanks for taking a look at this! I believe we actually removed this property from ActionButton intentionally. The plan was to implement a new component, `ToggleButton`, that included this state along with the associated toggle behavior. Along with the behavior differences, there's also some additional accessibility requirements for exposing the toggle state to assistive technology, so that's why we thought it would make more sense as a separate component.

Let me get back to you with the API we'd like to implement.

**Comment by devongovett** [comment]
Ok, here's what the API should look like:

```tsx
interface ToggleButtonProps extends ButtonProps {
  /**  Whether the element should be selected (controlled). */
  isSelected?: boolean,
  /** Whether the element should be selected (uncontrolled). */
  defaultSelected?: boolean,
  /** Handler that is called when the element's selection state changes. */
  onChange: (isSelected: boolean) => void
}

interface SpectrumToggleButtonProps extends ToggleButtonProps, SpectrumActionButtonProps {
   /** Whether the button should be displayed with an [emphasized style](https://spectrum.adobe.com/page/action-button/#Emphasis). */
  isEmphasized?: boolean
}
```

We'd also need a new hook in React Aria to implement the selection behavior. This hook should call `useButton` internally, and also add the `aria-pressed` prop to the `buttonProps` to indicate whether the toggle button is selected or not. In addition, it should handle `onPress` and toggle the selection state.

```tsx
interface AriaToggleButtonProps extends ToggleButtonProps, AriaBaseButtonProps {}

declare function useToggleButton(props: ToggleButtonProps): ButtonAria;
```

The state should be managed by the existing `useToggleState` hook, which is already used for checkboxes and switches, and will handle controlled/uncontrolled state and firing `onChange` appropriately.

**Comment by tomsontom** [comment]
Ok - i‘ll give that a try - is this PR the right place to proceed or should we close that one create a new ticket and start there from scratch?

**Comment by devongovett** [comment]
Either way works for us!

**Comment by tomsontom** [comment]
I've created #915 who also contains #907

**Pull request closed without merging**